docs/openapi/howto/archboard_feedback.md

```diff
@@ -0,0 +1,152 @@
+- [Operation should be pageable](#operation-should-be-pageable)
+
+## Need for more/less SDK package(s)
+
+If you got as a feedback that your API needs to split across several packages, or two packages should be merged into one, then
+structure your Swaggers to have exactly one folder per package. Most of the work is re-organizing files and Readme. Example of hierarchy on 
+the [RestAPI spec repo](https://github.com/Azure/azure-rest-api-specs):
+
+```
+specification/
+├─ myservice/
+│  ├─ data-plane/
+│  │  ├─ package1/
+│  │  │  ├─ Microsoft.MyService/ (swaggers for package 1)
+│  │  │  ├─ readme.md
+│  │  ├─ package2/
+│  │  │  ├─ Microsoft.MyService/ (swaggers for package 2)
+│  │  │  ├─ readme.md
+```
+
+## Rename an operation
+
+Operation name are directly OperationID in Swagger. Just rename the OperationID directly.
+
+Note: If your operation contains an `_`, just change the second part as the first is an operation group.
+
+``` json
+ "operationId": "SomethingToDoOperation",
+ ...
+ "operationId": "MyGroup_SomethingElseToDoOperation",
+```
+becomes
+``` json
+ "operationId": "BuildTheThing",
+ ...
+ "operationId": "MyGroup_BuildSomethingElse",
+```
+
+## Rename an operation group (or rename a sub-client)
+
+Operation groups are the first part of an OperationID in Swagger, before the `_` character. To rename an operation group, you need to rename all OperationID that
+starts with the operation group prefix (there is likely many of them).
+
+``` json
+ "operationId": "MyGroup_BuildSomethingElse",
+```
+becomes
+``` json
+ "operationId": "NewGroup_BuildSomethingElse",
+```
+
+## Need one client to have all operations directly on it (no sub-client)
+
+Don't use operation groups in your OperationID (remove them if they exist). For any operation with an `_` character, remove the first part.
+
+``` json
+ "operationId": "MyGroup_BuildSomethingElse",
+```
+becomes
+``` json
+ "operationId": "BuildSomethingElse",
+```
+
+## Need one client/builder with sub-clients
+
+-	Use operation groups, and name them as you want your subclients to be called. Operation group can be create in Swagger by using the `_` character in the OperationID 
+(`OperationGroupName_OperationName`)
+-	For C#, use `single-top-level-client: true`
+
+For instance, this:
+``` json
+ "operationId": "MyGroup_BuildSomethingElse",
+```
+
+Will create a subclient for MyGroup, that will have an operation called BuildSomethingElse.
+In various languages this could looks like (pseudo-code):
+```
+MyServiceClient(endpoint, credentials).MyGroup().BuildSomethingElse() // C#
+
+MyServiceClient(endpoint, credentials).my_group.build_something_else() # Python
+```
+
+## Need two or more service clients (regardless of if some needs subclients or not)
+
+-	Make sure you don't have one Swagger with operations that are designed to be in two different clients.
+  Split Swagger files in multiple swagger files if necessary: clients should correspond to a clear set of Swagger files.
```

Comment:
This seems to imply that one swagger file is always meant to represent one-top level client. Is it fair to always assume that will hold true?

Reply:
I think we want to encourage that as much as possible. Multiple swagger files introduce many complexities and potential for problems down the road -- like different api-versions.

Reply:
If I am understanding you correctly @mikekistler, there's value in trying as much as possible to have a single swagger file. The solution to the "Need two or more service clients" feedback in the doc is suggesting to create multiple swagger files. When is it a good idea to take on the complexities that multiple swagger files introduce?

> I think we want to encourage that as much as possible.

To clarify, by "that", are you referring to the one-swagger = one-client approach?

Reply:
Sorry for not being clear.  We want to encourage teams to use only one swagger to generate a client.  That is, we want to discourage teams from asking autorest to combine or merge multiple swagger files together and generate a single client from that.

Reply:
My main point was that if you have on Swagger, and you want two service clients, you need to split that Swagger in two. If you got the feedback that you need two service clients, it's because the scenario are really different.